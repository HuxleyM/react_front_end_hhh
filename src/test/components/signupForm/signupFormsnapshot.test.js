import React, { Component } from 'React';
import { create } from "react-test-renderer";
import SignUpForm from '../../../components/signupForm'
import {shallow, mount, render} from 'enzyme'

describe("SignupForm", () => {
  test("it matches the snapshot", () => {
    const component = create(<SignUpForm />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('Sign up', () => {

  const wrapper = mount(
    <SignUpForm/>
  )

  it('displays a form to signup with', () => {
    expect(wrapper.find(<form></form>)).toBeTruthy();
  });

  it('the form can take a venuename', () => {
    expect(wrapper.contains('Venue Name')).toBe(true)
  })

  it('the form can take an address', () => {
    expect(wrapper.contains('Venue Address')).toBe(true)
  })

  it('the form can take an email', () => {
    expect(wrapper.contains('Email')).toBe(true)
  })

  it('the form can take an password', () => {
    expect(wrapper.contains('Password')).toBe(true)
  })

})
