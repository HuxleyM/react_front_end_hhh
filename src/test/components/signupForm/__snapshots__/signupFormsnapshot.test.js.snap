// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm it matches the snapshot 1`] = `
<form>
  <label>
    Venue Name
  </label>
  <input
    id="venuename"
    placeholder="Crisis Cafe"
    type="text"
  />
  <label>
    Venue Address
  </label>
  <input
    id="venueaddress"
    placeholder="10 Commercial Street"
    type="text"
  />
  <label>
    Email
  </label>
  <input
    id="venueEmail"
    placeholder="venue@gmail.com"
    type="email"
  />
  <label>
    Password
  </label>
  <input
    id="password"
    placeholder="qwe123"
    type="password"
  />
  <input
    className="submitForm"
    id="submit_form"
    type="submit"
    value="Submit"
  />
</form>
`;
